api/v1alpha3/machinehealthcheck_types.go

```diff
@@ -84,6 +84,11 @@ type MachineHealthCheckStatus struct {
 	// +kubebuilder:validation:Minimum=0
 	CurrentHealthy int32 `json:"currentHealthy,omitempty"`
 
+	// number of further remediations allowed by this machine health check before
+	// maxUnhealthy short circuiting will be applied
+	// +kubebuilder:validation:Minimum=0
+	AllowedRemediations int32 `json:"allowedRemediations,omitempty"`
```

Comment:
Any other name that'd be suitable here? I find "Allowed" a little misleading, but I don't have a good suggestion

Reply:
Hmm, I'll have to have a think on that one, welcome to suggestions. I had taken the inspiration from the `allowedDisruptions` field on a PDB, I feel like this represents a very similar concept

Reply:
I think allowed makes sense, especially after seeing it in this context (from #3013)
```yaml
spec:
  maxUnhealthy: 40%
status:
  expectedMachines: 8
  currentHealthy: 6
  allowedRemediations: 1
```

Reply:
I think maybe "CurrentRemediationBudget" might make more sense.  I think this better confers to the user what the user has left. "AllowedRemediations" makes it look like we're over the limit already.

Overall, I'm not sure I like this field.  I think maybe a boolean of "RemediationsAllowed" would be clearer here, even if redundant to the status object.

Reply:
Personally I think it's useful to have a numeric field in the status for the number of further remediations that are allowed as it will give users an idea of how many machines the controller thinks can go further unhealthy, eg, on a percentage based system where rounding is involved, it shows which way the controller rounds. It may also be helpful tracking down bugs in the future, eg where `"10"` was treated as `"10%"` because of an error in the intorstring implementation, you could immediately see that the number the controller is interpreting is wrong.

I would prefer `RemediationBudget` to `CurrentRemediationBudget` as I feel the word `Current` is a bit superfluous, status fields are implicitly the "current" state

Reply:
`remediationsAllowed` makes slightly more sense to me, especially since that is a closer match to `disruptionsAllowed` https://kubernetes.io/docs/tasks/run-application/configure-pdb/#check-the-status-of-the-pdb